sql/src/main/java/org/apache/druid/sql/calcite/parser/DruidSqlParserUtils.java

```diff
@@ -566,14 +571,28 @@ public static String parseColumnName(SqlNode sqlNode)
    * @return the timestamp string as milliseconds from epoch
    * @throws DruidException if the SQL node is not a SqlTimestampLiteral
    */
-  private static String parseTimeStampWithTimeZone(SqlNode sqlNode, DateTimeZone timeZone)
+  static String parseTimeStampWithTimeZone(SqlNode sqlNode, DateTimeZone timeZone)
   {
+    Timestamp sqlTimestamp;
+    ZonedDateTime zonedTimestamp;
+
+    if (sqlNode instanceof SqlUnknownLiteral) {
+      try {
+        SqlTimestampLiteral timestampLiteral = (SqlTimestampLiteral) ((SqlUnknownLiteral) sqlNode).resolve(SqlTypeName.TIMESTAMP);
+        sqlTimestamp = Timestamp.valueOf(timestampLiteral.toFormattedString());
+      }
+      catch (Exception e) {
+        throw InvalidSqlInput.exception("Cannot get a timestamp from sql expression [%s]", sqlNode);
```

Comment:
can this be rephrased a bit since the end-user is going to see this error? I can suggest something if there is an example scenario where this error will pop up.

Reply:
I'll add an unit test for this